components/widgets/resources/src/main/resources/resources/uicomponents/widgets/extraGeneVariantData.js

```diff
@@ -353,6 +353,18 @@ var ExtraGeneVariantData = (function (ExtraGeneVariantData) {
       newMoreInfoRow.insert(varMoreInfoInner);
       variantFooter.insert({before : newMoreInfoRow});
 
+      // restrict only numerical input for start_position and end_position variant fiellds
+      newMoreInfoRow.select('input:[id$=_start_position]', 'input:[id$=_end_position]').each( function(input) {
+        input.observe('keyup', function(event){
+          this.value=this.value.replace(/[^\d]/,'');
+        });
+      });
```

Comment:
Problems with this:
1/ I can paste any text, not just numbers, and only the first non-digit characters will be removed. This is a big issue because:
* The users are likely to copy & paste positions from a lab report rather than type them
* It's easy to accidentally copy some extra spaces around the number
* A numerical value with spaces around it unfortunately is not parsed as a number and not saved in the number field, so the data will be lost upon saving.

2/ If I hold a letter key to type multiple letters, it will only remove the first one.

Both can probably  be fixed if you do a global replace: `this.value = this.value.replace(/[^\d]/g,'');`.